docs/Routes.md

```diff
@@ -170,14 +170,33 @@ fastify.get('/', options, async function (request, reply) {
 ```
 **Warning:** You can't return `undefined`. For more details read [promise-resolution](#promise-resolution).
 
-As you can see we are not calling `reply.send` to send back the data to the user. You just need to return the body and you are done!
+<a name="callback"></a>
+### Callback
+
+As you see [above](/#async-await) we are not calling `reply.send` to send back the data to the user. You just need to return the body and you are done!
 
 If you need it you can also send back the data to the user with `reply.send`.
 ```js
-fastify.get('/', options, async function (request, reply) {
-  var data = await getData()
-  var processed = await processData(data)
-  reply.send(processed)
+fastify.get('/', options, function (request, reply) {
+  getData()
+  .then((data) => processData(data))
+  .then((processed) => reply.send(processed))
+  .catch((err) => {
+    reply.code(500).send(err)
+  })
+})
+```
+
+<a name="promises"></a>
+### Promises
+
+As you see [above](/#callback) the code is very verbose because we combine different control-flow styles (`callback-style` and `async-await`) and error-handling is explicit. We can fix it very easily:
+
+If you **return** the promise the value is returned to the user and error-handling is handled by the framework.
+```js
+fastify.get('/', options, function (request, reply) {
```

Comment:
```suggestion
fastify.get('/', options, async function (request) {
```

Reply:
I think it was intended not to use an async function here. 
There is an async/await example starting at line 162.

Reply:
Exactly, that example should demonstrate the promise only approach.